src/hexkit/providers/akafka/testutils.py

```diff
@@ -408,71 +406,50 @@ def _build_record_deletion_command(self, delete_config: JsonObject) -> str:
         )
         return f"{echo_command} && {deletion_command}"
 
-    def clear_topics(
+    async def _get_topic_description(
         self,
-        *,
-        topics: Optional[Union[str, list[str]]] = None,
-    ):
-        """
-        Clear messages from given topic(s).
+        topics: Optional[Union[list[str], str]] = None,
+        exclude_internal: bool = True,
+    ) -> JsonObject:
+        """Get a decription of the given topic(s).
 
-        When no topics are specified, all existing topics will be cleared.
+        If no topics are specified, all topics will be covered, except internal topics
+        unless otherwise specified.
         """
-        admin_client = KafkaAdminClient(bootstrap_servers=self.kafka_servers)
+        admin_client = AIOKafkaAdminClient(bootstrap_servers=self.kafka_servers)
+        await admin_client.start()
         try:
-            all_topics = admin_client.list_topics()
-
-            # If not clearing specific topics, delete all aside from the internal
-            # __consumer_offsets topic
             if topics is None:
-                topics = [
-                    topic for topic in all_topics if topic != "__consumer_offsets"
-                ]
+                if exclude_internal:
+                    topics = [
+                        topic
+                        for topic in await admin_client.list_topics()
+                        if not topic.startswith("__")
+                    ]
```

Comment:
```suggestion
                    ]
                else:
                    topics = await admin_client.list_topics()
```
Docstring is ambiguous here for  `exclude_internal == False`.
Currently topics would be `None` in this case.
Is this intended?

Reply:
Yes, the admin client understands that None = all.

Reply:
Added a comment to explain this now.